pkg/api/defaults-kubelet.go

```diff
@@ -72,7 +72,7 @@ func (cs *ContainerService) setKubeletConfig(isUpgrade bool) {
 	staticWindowsKubeletConfig["--cloud-config"] = "c:\\k\\azure.json"
 	staticWindowsKubeletConfig["--cgroups-per-qos"] = "false"
 	staticWindowsKubeletConfig["--enforce-node-allocatable"] = "\"\"\"\""
-	staticWindowsKubeletConfig["--system-reserved"] = "memory=2Gi"
+	staticWindowsKubeletConfig["--system-reserved"] = "memory=2Gi,cpu=500m"
```

Comment:
How much is the kubelet consuming currently? If system components + kubelet is consuming more than 500mm won't it be a problem? Shouldn't we add `--kube-reserved` too considering those metrics are collected from kubelet?

Reply:
> How much is the kubelet consuming currently? 

We don't have these numbers currently across different types of nodes and deployment loads currently. Though through some early testing we did see we need to scale this number as the node size grows.

> If system components + kubelet is consuming more than 500mm won't it be a problem? 

Yes if the node allocatable is fully consumed and kubelet goes over the reserve there is potential for this to occur.  This idea was in the [original design](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/node-allocatable.md#kubernetes-reservation-is-smaller-than-kubernetes-component-usage).  It is also the case for other components that run on the node like CRI and CNI.  We are also looking into adding a flag to kubelet that will give kubelet higher priority in cases such as this.  

> --kube-reserved too considering those metrics are collected from kubelet

In Windows I believe `--system-reserve` and `--kubelet-reserve` only subtract from Allocatable and don't have any of the enforcements or concepts of creating cgroups in place that Linux does via `--enforce-node-allocatable` or `--cgroups-per-qos`.  

In this case it is a matter of semantics.  We are attempting to reserve for not only kubelet but other system services and the system services are not grouped into jobs (similiar to cgroups) so chose system-reserve.

Reply:
> In Windows I believe --system-reserve and --kubelet-reserve only subtract from Allocatable and don't have any of the enforcements or concepts of creating cgroups in place that Linux does via --enforce-node-allocatable or --cgroups-per-qos.

Ahh I see, in case of OpenShift we combine --system-reserve and --kube-reserve to map to a single cgroup, effectively same as subtracting the value from node-allocatable.

> In this case it is a matter of semantics. We are attempting to reserve for not only kubelet but other system services and the system services are not grouped into jobs (similiar to cgroups) so chose system-reserve.

That makes sense.

> Though through some early testing we did see we need to scale this number as the node size grows.

Interesting, do you have a dedicated scale testing environment?